src/main/java/net/minecraftforge/items/IItemHandlerIterator.java

```diff
@@ -75,19 +76,52 @@
      *
      * @return the previous element in the list
      * @throws NoSuchElementException if the iteration has no previous
-     *         element
+     *                                element
      */
     @Nonnull
     ItemStack previous();
 
+
+    /**
+     * @param stack to try insert into the index it is current at
+     * @return the Remainder that was not inserted
+     */
+    @Nonnull
+    ItemStack set(@Nonnull ItemStack stack);
```

Comment:
Can we have some sort of insertion result here?

Reply:
like ??